connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/Plugins.java

```diff
@@ -0,0 +1,235 @@
+                        return new DelegatingClassLoader(paths);
+                    }
+                }
+        );
+    }
+
+    public DelegatingClassLoader delegatingLoader() {
+        return delegatingLoader;
+    }
+
+    public Map<PluginClassLoader, URL[]> getLoaders() {
+        return loaders;
+    }
+
+    public Set<PluginDesc<Connector>> connectors() {
+        return delegatingLoader.connectors();
+    }
+
+    public Set<PluginDesc<Converter>> converters() {
+        return delegatingLoader.converters();
+    }
+
+    public Set<PluginDesc<Transformation>> transformations() {
+        return delegatingLoader.transformations();
+    }
+
+    @SuppressWarnings("unchecked")
+    public Connector newConnector(String connectorClassOrAlias) {
+        Class<? extends Connector> klass;
+        try {
+            klass = pluginClass(
+                    delegatingLoader,
+                    connectorClassOrAlias,
+                    Connector.class
+            );
+        } catch (ClassNotFoundException e) {
+            List<PluginDesc<Connector>> matches = new ArrayList<>();
+            for (PluginDesc<Connector> plugin : delegatingLoader.connectors()) {
+                Class<?> pluginClass = plugin.pluginClass();
+                String simpleName = pluginClass.getSimpleName();
+                if (simpleName.equals(connectorClassOrAlias)
+                        || simpleName.equals(connectorClassOrAlias + "Connector")) {
+                    matches.add(plugin);
+                }
+            }
+
+            if (matches.isEmpty()) {
+                throw new ConnectException(
+                        "Failed to find any class that implements Connector and which name matches "
+                        + connectorClassOrAlias
+                        + ", available connectors are: "
+                        + pluginNames(delegatingLoader.connectors())
+                );
+            }
+            if (matches.size() > 1) {
+                throw new ConnectException(
+                        "More than one connector matches alias "
+                        + connectorClassOrAlias
+                        +
+                        ". Please use full package and class name instead. Classes found: "
+                        + pluginNames(matches)
+                );
+            }
+
+            PluginDesc<Connector> entry = matches.get(0);
+            delegatingLoader.addAlias(entry, connectorClassOrAlias);
+            klass = entry.pluginClass();
+        }
+        return newPlugin(klass);
+    }
+
+    private static <T> String pluginNames(Collection<PluginDesc<T>> plugins) {
+        StringBuilder names = new StringBuilder();
+        for (PluginDesc<T> plugin : plugins) {
+            names.append(plugin.className()).append(", ");
+        }
+        String result = names.toString();
+        return result.isEmpty()
+               ? ""
+               : result.substring(0, result.length() - 2);
```

Comment:
There's a `join` utility in the `Utils` class in common that will get rid of this logic and extra `substring` operation.